testing/scenario_app/lib/src/platform_view.dart

```diff
@@ -26,21 +27,164 @@ List<int> _to64(num value) {
 }
 
 /// A simple platform view.
-class PlatformViewScenario extends Scenario {
+class PlatformViewScenario extends Scenario
+    with _BasePlatformViewScenarioMixin {
   /// Creates the PlatformView scenario.
   ///
   /// The [window] parameter must not be null.
   PlatformViewScenario(Window window, String text, {int id = 0})
       : assert(window != null),
         super(window) {
+    constructScenario(window, text, id);
+  }
+
+  @override
+  void onBeginFrame(Duration duration) {
+    final SceneBuilder builder = SceneBuilder();
+
+    builder.pushOffset(0, 0);
+
+    finishBuilderByAddingPlatformViewAndPicture(builder, 0);
+  }
+}
+
+/// Platform view with clip rect.
+class PlatformViewClipRectScenario extends Scenario
+    with _BasePlatformViewScenarioMixin {
+  /// Constuct a platform view with clip rect scenario.
+  PlatformViewClipRectScenario(Window window, String text, {int id = 0})
+      : assert(window != null),
+        super(window) {
+    constructScenario(window, text, id);
+  }
+
+  @override
+  void onBeginFrame(Duration duration) {
+    final SceneBuilder builder = SceneBuilder();
+    builder.pushOffset(0, 0);
+    builder.pushClipRect(const Rect.fromLTRB(100, 100, 400, 400));
+    finishBuilderByAddingPlatformViewAndPicture(builder, 1);
+  }
+}
+
+/// Platform view with clip rrect.
+class PlatformViewClipRRectScenario extends PlatformViewScenario {
+  /// Constuct a platform view with clip rrect scenario.
+  PlatformViewClipRRectScenario(Window window, String text, {int id = 0})
+      : super(window, text, id: id);
+
+  @override
+  void onBeginFrame(Duration duration) {
+    final SceneBuilder builder = SceneBuilder();
+
+    builder.pushOffset(0, 0);
+    builder.pushClipRRect(
+      RRect.fromLTRBAndCorners(
+        100,
+        100,
+        400,
+        400,
+        topLeft: const Radius.circular(15),
+        topRight: const Radius.circular(50),
+        bottomLeft: const Radius.circular(50),
+      ),
+    );
+    finishBuilderByAddingPlatformViewAndPicture(builder, 2);
+  }
+}
+
+/// Platform view with clip path.
+class PlatformViewClipPathScenario extends PlatformViewScenario {
+  /// Constuct a platform view with clip rrect scenario.
+  PlatformViewClipPathScenario(Window window, String text, {int id = 0})
+      : super(window, text, id: id);
+
+  @override
+  void onBeginFrame(Duration duration) {
+    final SceneBuilder builder = SceneBuilder();
+
+    builder.pushOffset(0, 0);
+
+    // Create a path of rectangle with width of 200 and height of 300, starting from (100, 100).
+    //
+    // The left side of the rectangle becomes a symmetric curve towards the left.
+    // The right side of the rectangle becomes a double curve. From top of bottom of the curve, it goes left then right.
+    //     _______
+    //    |      |
```

Comment:
Nit: Maybe just a reference to the golden image for this, or a link to it on github instead of the ASCII art.